src/NFTBill.sol

```diff
@@ -40,15 +40,20 @@ contract NFTBill is ERC1155 {
     }
 
     function withdraw(uint256 id) external {
+        withdrawTo(msg.sender, id);
+    }
+
+    // withdraw token to specified address, save gas fee on calling NFT transfer method
+    function withdrawTo(address to, uint256 id) public {
         _burn(msg.sender, id, 1);
```

Comment:
Shouldn't this be burning the `to` address?

Reply:
The NFT owner is the msg.sender, but withdraw the coin to another address.

Reply:
from my understanding this is how it will work:

1. NFT owner provides a signature to relayer
2. Relayer will call our permit-like function using that signature: https://github.com/enclosed-money/contracts/blob/77bca2761e47e6fb0f61eb33e75a958e4f9afef3/src/NFTBill.sol#L57
3. Relayer will then have permission to call the withdrawTo function. This means the msg.sender is the relayer's address not the owner's

Reply:
Ah I see you are thinking about a different use case